opensearch-operator/pkg/builders/cluster.go

```diff
@@ -1105,6 +1118,19 @@ func NewServiceMonitor(cr *opsterv1.OpenSearchCluster) *monitoring.ServiceMonito
 		},
 	}
 
+	if cr.Spec.General.Monitoring.Scheme == "" {
+		cr.Spec.General.Monitoring.Scheme = "https"
+	}
+
+	safetlsconfig := monitoring.SafeTLSConfig{
+		ServerName:         cr.Spec.General.Monitoring.TLSConfig.ServerName,
+		InsecureSkipVerify: cr.Spec.General.Monitoring.TLSConfig.InsecureSkipVerify,
+	}
+
+	tlsconfig := monitoring.TLSConfig{
```

Comment:
Shouldn't this struct only be initialized if the fields are actually set in the CRD and otherwise use nil?

Reply:
@rhysxevans This comment is not yet addressed. Everything else looks good.

Reply:
Hi, I am going to ask a newb question as I am one, how would I go about doing that? If it is not defined at runtime it gets setup but with the default empty values

Reply:
Any help is appreciated

Reply:
In the types you'll have to write it as `TLSConfig  *MonitoringConfigTLS` (the asterix to make it a pointer),

then in the code you can do something like
```go
var tlsconfig *monitoring.TLSConfig
if cr.Spec.General.Monitoring.TLSConfig != nil {
    // ...
    tlsconfig = &monitoring.TLSConfig{
        SafeTLSConfig: safetlsconfig,
    }
} else {
    tlsconfig = nil
}
// ...
return &monitoring.ServiceMonitor{
    Spec: monitoring.ServiceMonitorSpec{
        Endpoints: []monitoring.Endpoint{
            {Port: "http",
                // ...
                TLSConfig: tlsconfig,
            },
        },
    },
}
```

Reply:
Hi @swoehrl-mw, I have attempted to fix the code, please let me know if it is ok or not. Thanks

Reply:
Looks good